### PR TITLE
Alt Click 2 Rotate Pipes

### PR DESCRIPTION
**Pull request by kugamo**
## What Does This PR Do

Adds ability to rotate/flip pipes when AltClick or AltShiftClick on pipes.

## Why It's Good For The Game

QOL, skip the right click menu and embrace the hotkey lifestyle.

## Changelog
:cl:
add: Alt clicking on pipes will now rotate them, Alt+Shift clicking will flip them.
/:cl:

### CODE REVIEW

File: code/game/machinery/pipe/construction.dm [hunk 1]
```diff
@@ -560,3 +560,13 @@
 		our_rpd.delete_single_pipe(user, src)
 	else
 		..()
+
+/obj/item/pipe/AltClick(mob/user)
+	. = ..()
+	if(user.Adjacent(src))
```
Comment (Sirryan2002) [inline code/game/machinery/pipe/construction.dm:566]: can you test to make sure this doesn't allow observers to rotate pipes?

**Review by hal9000PR — DISMISSED** [review]
would you add these changes to the examine proc of pipes? so people who can examine the pipe and instantly know they can altclick/altshiftclick to rotate/flip.

File: code/game/machinery/pipe/construction.dm [hunk 2]
```diff
@@ -560,3 +563,14 @@
...
+/obj/item/pipe/AltClick(mob/user)
+	if(user.incapacitated() || !(user.Adjacent(src)))
+		return
+	rotate()
+
+/obj/item/pipe/AltShiftClick(mob/user)
+	. = ..()
+	if(user.incapacitated() || !(user.Adjacent(src)))
+		return
+	flip()
```
Comment (hal9000PR) [inline code/game/machinery/pipe/construction.dm:576]:
```suggestion
/obj/item/pipe/AltClick(mob/user)
	if(user.incapacitated() || !user.Adjacent(src))
		return
	rotate()

/obj/item/pipe/AltShiftClick(mob/user)
	. = ..()
	if(user.incapacitated() || !user.Adjacent(src))
		return
	flip()
```

File: code/game/machinery/pipe/construction.dm [hunk 3]
```diff
@@ -560,3 +563,14 @@
...
 	else
 		..()
+
+/obj/item/pipe/AltClick(mob/user)
+	if(user.incapacitated() || !user.Adjacent(src))
+		return
+	rotate()
+
+/obj/item/pipe/AltShiftClick(mob/user)
+	. = ..()
```
Comment (SteelSlayer) [inline code/game/machinery/pipe/construction.dm:573]: Since the parent does nothing.
```suggestion
```

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**